service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/NotifyGatekeeperControllerAboutToStartTest.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(NotifyGatekeeperController.class)
+@OverrideAutoConfiguration(enabled = true)
+class NotifyGatekeeperControllerAboutToStartTest extends AbstractControllerTest {
+
+    private static final String SUBMITTED = "Submitted";
+    private static final String GATEKEEPING = "Gatekeeping";
+
+    NotifyGatekeeperControllerAboutToStartTest() {
+        super("notify-gatekeeper");
+    }
+
+    @Test
+    void shouldReturnErrorsWhenFamilymanNumberIsNotProvided() {
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails("", SUBMITTED));
+
+        assertThat(callbackResponse.getErrors()).containsExactly("Enter Familyman case number");
+    }
+
+    @Test
+    void shouldNotValidateFamilymanNumberWhenInGatekeepingState() {
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails("", GATEKEEPING));
+
+        assertThat(callbackResponse.getErrors()).isEmpty();
```

Comment:
As per method name, don't we need to check for method call details here "validateGroupService.validateGroup"?

Reply:
That would probably be a better way of doing it, I just ensured that bad data was sent but I think maybe spying on the validator would be better.